#!/usr/bin/python
# coding=utf-8
import argparse
import os.path
import sys
import traceback

sys.path = ([sys.path[0]]
            + [os.path.join(os.path.dirname(__file__), "../lib")]
            + sys.path[1:])
import dataprocessor as dp
sys.path = [sys.path[0]] + sys.path[2:]


def main():
    parser = argparse.ArgumentParser(description="""
                command line interface for DataProcessor pipeline""")
    parser.add_argument('manip_json')
    parser.add_argument('-d', "--data", help="The path of data JSON")
    args = parser.parse_args()

    json_fn = dp.utility.check_file(args.manip_json)
    if args.data:
        data_fn = dp.utility.path_expand(args.data)
        with open(json_fn, 'r') as f, dp.io.SyncDataHandler(data_fn,
                                                            duration=1) as dh:
            dp.execute.execute_from_json_str(f.read(), dh.get())
    else:
        with open(json_fn, 'r') as f:
            dp.execute.execute_from_json_str(f.read())


if __name__ == "__main__":
    try:
        main()
    except dp.exception.InvalidJSONError as e:
        print("ERROR found in your JSON")
        print("Pipe name: %s" % e.name)
        print("Message  : %s" % e.msg)
        sys.exit(1)
    except dp.exception.DataProcessorError as e:
        print(traceback.format_exc())
        print("ERROR in command-line arguments.")
        print("Message: %s" % e.msg)
        sys.exit(1)
